textadept:11.4 -> 12.0

mhwombat opened the pull request:
###### Description of changes

This is for issue #214776.

The changes are described at https://github.com/orbitalquark/textadept/discussions/300, but here's a summary.

- Going forward, Qt will be the default GUI toolkit for Textadept on Windows, macOS, and Linux.
- The GTK version of Textadept is still supported, but only on Linux. If you want to build with GTK, you will need to do so manually; binaries will no longer be provided.
- Textadept has switched to a CMake-based build for building natively on Windows, macOS, and Linux.

I have modified the textadept package to reflect these changes. (The code has a few TODOs for things I will change once the new version is actually released.

In updating the package, I have switched over completely to Qt. There's no support for GTK in my updated package. If anyone really needs GTK, I will need time to figure out how to build it, and to find a way to test it. I also didn't include a way to build a curses version of Textadept. Again, if anyone needs that, it will take me time to figure out how to build and test it. _This is why I'm doing a premature pull request; to get feedback on these issues._

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

mhwombat committed:
Added TODO messages in comments

7c6f434c commented on pkgs/top-level/aliases.nix:
```diff
@@ -1487,7 +1487,8 @@ mapAliases ({
   tex-gyre-termes-math = throw "'tex-gyre-termes-math' has been renamed to/replaced by 'tex-gyre-math.termes'"; # Converted to throw 2022-02-22
-  textadept11 = textadept; # Added 2022-06-07
+  textadept11 = throw "textadept11 has been removed. Please use textadept12 instead"; # Added 2022-12-23 TODO: UPDATE THE DATE
+  textadept12 = textadept; # Added 2022-12-23 TODO: UPDATE THE DATE
```
If we give up on keeping even the previous version, maybe _not_ add `textadept12` alias? `textadept11` alias was added when dropping the 10/11 two-versions support.

mhwombat replied:
Thank you for the feedback. So I should remove line 1491. What about line 1490 -- should I put that back the way it was?

7c6f434c replied:
I think it's fine to replace it with a throw (as we will have no TA 11 afterwards) with advice to use `textadept` the non-version-pinned.

mhwombat replied:
done!

winterqt commented:
This needs to be split up into more descriptive comments, it seems like you're doing a lot more than the comment additions :)

mhwombat commented:
> This needs to be split up into more descriptive comments, it seems like you're doing a lot more than the comment additions :)

Sorry, the commit message is just from my most recent commit to my repo, which only affected comments.

Here's a summary of the changes I made:

1. I modified `default.nix` to use CMAKE instead of Make. This is required for the new version of Textadept.
2. I also removed the support for a curses version of Textadept from `default.nix`.
3. I updated `deps.nix` with the new dependencies.
4. Since the new version of Textadept uses QT instead of GTK, I modified `all-packages.nix` to use `libsForQt5.callPackage` instead of `callPackage`.
5. I added an alias to `aliases.nix`, but seems that should be removed.

mhwombat committed:
textadept:11.4 -> 12.0

winterqt commented:
Looks like this should be marked as a draft, so I've gone ahead and done so.

mhwombat committed:
textadept:11.4 -> 12.0

mhwombat committed:
textadept:11.4 -> 12.0

mhwombat committed:
textadept:11.4 -> 12.0

mhwombat committed:
textadept:11.4 -> 12.0
